public/New-DbaDbUser.ps1

```diff
@@ -69,165 +74,246 @@ function New-DbaDbUser {
         [Parameter(Mandatory, ParameterSetName = "Login")]
         [string]$Login,
-        [Parameter(ParameterSetName = "Login", Mandatory = $False)]
-        [Parameter(Mandatory, ParameterSetName = "NoLogin")]
-        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
-        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
-        [string]$Username,
-        [string]$DefaultSchema = 'dbo',
-        [Parameter(ParameterSetName = "ContainedSQLUser", Mandatory = $True)]
-        [securestring]$Password,
-        [Parameter(ParameterSetName = "ContainedAADUser", Mandatory = $True)]
+        ### If we need to pass a password to the command, we always use the type securestring and name the parameter SecurePassword. Here we only use the alias for backwords compatility.
+        [Parameter(Mandatory, ParameterSetName = "ContainedSQLUser")]
+        [Alias("Password")]
+        [securestring]$SecurePassword,
+        [Parameter(Mandatory, ParameterSetName = "ContainedAADUser")]
         [switch]$ExternalProvider,
+        ### Now we add parameters to specify the individual attributes for the object that are not specific for a parameter set.
+        ### As we want to use the schema dbo in most cases, we use default values in those cases.
+        [string]$DefaultSchema = 'dbo',
+        ### All commands that create new objects have a switch parameter Force to drop and re-create the object in case it already exists.
+        ### Sometimes, this parameter also changes the ConfirmPreference to "none". This way, -WhatIf and -Force cannot be used together. So this has to be removed everywhere.
+        ### This parameter is always the second last parameter.
         [switch]$Force,
+        ### All public commands have a switch parameter called EnableException as the last parameter. This changes the behavior of Stop-Function inside of the command.
         [switch]$EnableException
     )
 
     begin {
-        $connParam = @{ }
-        if ($SqlCredential) { $connParam.SqlCredential = $SqlCredential }
-
-        if ($Force) { $ConfirmPreference = 'none' }
+        ### To help analysing bugs in commands using parameter sets, we write the used parameter set to verbose output.
+        Write-Message -Level Verbose -Message "Using parameter set $($PSCmdlet.ParameterSetName)."
+
+        ### In case we don't use parameter sets, we check valid combinations of parameters here.
+        ### Option 1: Just write a warning message
+        ### Option 2: Use Stop-Function to stop the command.
+        ### Important for Option 2 to stop the command in case EnableException is $false:
+        ### * Add "return" to stop the execution of the begin block.
+        ### * Add "if (Test-FunctionInterrupt) { return }" to the process and the end block to stop the execution of the command.
+        ### Here we use Option 1 (Discussion: I would prefere this):
```

Comment:
"discussion" here is "we want to discuss it here" before committing these changes in the repo ?

Reply:
Yes, that was my intention.

Reply:
Removed the discussion as not needed anymore.